core/slim_sim.cpp

```diff
@@ -9353,18 +9354,33 @@ EidosValue_SP SLiMSim::ExecuteMethod_treeSeqRememberIndividuals(EidosGlobalStrin
 	if ((executing_block_type_ == SLiMEidosBlockType::SLiMEidosFitnessCallback) || (executing_block_type_ == SLiMEidosBlockType::SLiMEidosMateChoiceCallback) || (executing_block_type_ == SLiMEidosBlockType::SLiMEidosModifyChildCallback) || (executing_block_type_ == SLiMEidosBlockType::SLiMEidosRecombinationCallback))
 		EIDOS_TERMINATION << "ERROR (SLiMSim::ExecuteMethod_treeSeqRememberIndividuals): treeSeqRememberIndividuals() may not be called from inside a fitness(), mateChoice(), modifyChild(), or recombination() callback." << EidosTerminate();
 	
-	std::vector<const Genome *> genomes;
-	
 	for (int ind_index = 0; ind_index < ind_count; ind_index++)
 	{
-		Individual *ind = (Individual *)(individuals_value->ObjectElementAtIndex(ind_index, nullptr));
-		
-		genomes.emplace_back(ind->genome1_);
-		genomes.emplace_back(ind->genome2_);
+		Individual *individual = (Individual *)(individuals_value->ObjectElementAtIndex(ind_index, nullptr));
+        slim_pedigreeid_t ped_id = individual->pedigree_id_;
+        
+        // add this individual to the individual table if not present already
+        // NOTE: this means if an individual is remembered more than once, subsequent times have no effect
+        //  ... which I think is OK but should be noted in the documentation
```

Comment:
>  I don't see that that is an issue at all; like, what would it even mean for it to be otherwise? Remembering an individual is like setting a flag, right? Once the flag is set, setting it again makes no difference; flags can't be "double-set". So in my opinion you can just delete that "NOTE:" comment entirely, I think it's obvious.

Recall that we store mutable things like `age` and `location`. 

And, I think I disagree with my former self -- a more consistent behavior would be to store the state at the *last* time an individual was Remembered. For one thing, it's wierd for subsequent Remembers to have no effect; for another, if an individual is both Remembered and alive, then what gets output is the alive version, which is necessary to allow recovering from the saved state.

Reply:
Ah, I misunderstood what you were talking about here.  So there is, we agree, no such thing as being "double-remembered" as such; the point is just that each time an individual is remembered its information in the table should be updated to the current value (or not).  I agree with your current self: the information ought to be updated, yes.

Reply:
Some time you should show me how you manage to associate discussion threads with particular places in the code of the pull request, that's nice.  :->

Reply:
In "files changed", hovering over a line of code makes a blue plus mark pop up; if so, when you click you can make a comment there.